SU2_CFD/include/numerics/turbulent/turb_sources.hpp

```diff
@@ -28,331 +26,567 @@
+  CSourceBase_TurbSA(unsigned short nDim, unsigned short, const CConfig* config)
+      : CNumerics(nDim, 1, config),
+        idx(nDim, config->GetnSpecies()),
+        rotating_frame(config->GetRotating_Frame()),
+        transition(config->GetKind_Trans_Model() == TURB_TRANS_MODEL::BC) {
+    /*--- Setup the Jacobian pointer, we need to return su2double** but we know
+     * the Jacobian is 1x1 so we use this trick to avoid heap allocation. ---*/
+    Jacobian_i = &Jacobian_Buffer;
+  }
 
   /*!
-   * \brief Residual for source term integration.
-   * \param[in] intermittency_in - Value of the intermittency.
+   * \brief Get the intermittency for the BC transition model.
+   * \return Value of the intermittency.
    */
-  inline void SetIntermittency(su2double intermittency_in) final { intermittency = intermittency_in; }
+  su2double GetGammaBC() const final { return Gamma_BC; }
 
   /*!
    * \brief Residual for source term integration.
-   * \param[in] val_production - Value of the Production.
+   * \param[in] config - Definition of the particular problem.
+   * \return A lightweight const-view (read-only) of the residual/flux and Jacobians.
    */
-  inline void SetProduction(su2double val_production) final { Production = val_production; }
+  ResidualType<> ComputeResidual(const CConfig* config) override {
+    const auto& density = V_i[idx.Density()];
+    const auto& laminar_viscosity = V_i[idx.LaminarViscosity()];
+
+    AD::StartPreacc();
+    AD::SetPreaccIn(density, laminar_viscosity, StrainMag_i, ScalarVar_i[0], Volume, dist_i, roughness_i);
+    AD::SetPreaccIn(Vorticity_i, 3);
+    AD::SetPreaccIn(PrimVar_Grad_i + idx.Velocity(), nDim, nDim);
+    AD::SetPreaccIn(ScalarVar_Grad_i[0], nDim);
 
-  /*!
-   * \brief Residual for source term integration.
-   * \param[in] val_destruction - Value of the Destruction.
-   */
-  inline void SetDestruction(su2double val_destruction) final { Destruction = val_destruction; }
+    /*--- Common auxiliary variables and constants of the model. ---*/
+    CSAVariables var;
 
-  /*!
-   * \brief Residual for source term integration.
-   * \param[in] val_crossproduction - Value of the CrossProduction.
-   */
-  inline void SetCrossProduction(su2double val_crossproduction) final { CrossProduction = val_crossproduction; }
+    Residual = 0.0;
+    Jacobian_i[0] = 0.0;
 
-  /*!
-   * \brief ______________.
-   */
-  inline su2double GetProduction(void) const final { return Production; }
+    /*--- Evaluate Omega with a rotational correction term. ---*/
 
-  /*!
-   * \brief  Get the intermittency for the BC trans. model.
-   * \return Value of the intermittency.
-   */
-  inline su2double GetGammaBC(void) const final { return Gamma_BC; }
+    Omega::get(Vorticity_i, nDim, PrimVar_Grad_i + idx.Velocity(), var);
 
-  /*!
-   * \brief  ______________.
-   */
-  inline su2double GetDestruction(void) const final { return Destruction; }
+    /// TODO: Make this one of the template parameters?
```

Comment:
PR's are not easy to search either, I'll link back to this.

Reply:
Since the plan is to close this pull request once it is merged, writing a Todo's list at the end does not seem to be the best approach (In my opinion). I would suggest to include such a Todo's list in #1487 under the Turbulence models section.

Besides, the complementary notes from @pcarruscag inside the code can be helpful for future PR.